src/HindBrain/HindBrain.ino

```diff
@@ -0,0 +1,374 @@
+  // Update input global var with msg only if msg is 1 char.
+  if (strlen(msg.data) == 1) {
+    usrMsg = *msg.data;
+    #ifdef DEBUG
+    char j[20];
+    snprintf(j, sizeof(j), "Received %s", msg.data);
+    nh.loginfo(j);
+    #endif
+  }
+}
+
+
+void waitForUserVerification() {
+  // Block main code until 'y' is received on /user_input.
+  while (true) {
+    auto cmd = checkUserInput();
+    if (cmd == 'y') {
+      break;
+    }
+    nh.spinOnce();
+    delay(1);
+  }
+}
+
+
+void runStartupSequence() {
+  // Run startup & calibration sequence with appropriate user input.
+  //TODO create prompt publishing topic
+
+  nh.loginfo("Remove pins on steering and velocity, then publish 'y' to /user_input topic");
+  waitForUserVerification();
+
+  nh.loginfo("Verification received, homing steering and velocity actuators . . .");
+  delay(500);
+  rc1.SpeedAccelDeccelPositionM1(RC1_ADDRESS, 0, 300, 0, velMsg, 1);
+  rc1.SpeedAccelDeccelPositionM2(RC1_ADDRESS, 0, 500, 0, steerMsg, 1);
+  rc2.SpeedAccelDeccelPositionM2(RC2_ADDRESS, 0, 300, 0, hitchMsg, 1);
+  delay(500);
+
+  nh.loginfo("Re-install pins on steering and velocity, then publish 'y' to /user_input topic");
+  waitForUserVerification();
+  nh.loginfo("Verification received, vehicle ready to run.");
+}
+
+
+int steerAckToCmd(float ack){
+  // Return RoboClaw command given ackermann steering msg.
+  float cmd;
+
+  // Convert from input message to output command
+  if (ack > STEER_MSG_CENTER) {
+    cmd = map(ack, STEER_MSG_CENTER, STEER_MSG_LEFT, STEER_CMD_CENTER, STEER_CMD_LEFT);
+  } else if (ack < STEER_MSG_CENTER) {
+    cmd = map(ack, STEER_MSG_RIGHT, STEER_MSG_CENTER, STEER_CMD_RIGHT, STEER_CMD_CENTER);
+  } else {
+    cmd = STEER_CMD_CENTER;
+  }
+
+  // Safety limits for signal
+  if (cmd < STEER_CMD_LEFT) {
+    cmd = STEER_CMD_LEFT;
+    nh.logwarn("ERR: oversteering left");
+  }
+  else if (cmd > STEER_CMD_RIGHT) {
+    cmd = STEER_CMD_RIGHT;
+    nh.logwarn("ERR: oversteering right");
+  }
+
+  return cmd;
+}
+
+
+int velAckToCmd(float ack){
+  // Return RoboClaw command given ackermann velocity msg.
+  float cmd;
+
+  // Convert from range of input signal to range of output signal
+  cmd = map(ack, VEL_MSG_MIN, VEL_MSG_MAX, VEL_CMD_MIN, VEL_CMD_MAX);
+
+  // Safety limits for signal; feels switched bc high vals = low speed
+  if (cmd < VEL_CMD_MAX) {
+    cmd = VEL_CMD_MAX;
+  }
+  else if(cmd > VEL_CMD_MIN) {
+    cmd = VEL_CMD_MIN;
+  }
+
+  return cmd;
+}
+
+
+void checkSerial(ros::NodeHandle *nh) {
+  // Given node, estops if watchdog has timed out
+  // https://answers.ros.org/question/124481/rosserial-arduino-how-to-check-on-device-if-in-sync-with-host/
+
+  if(millis() - watchdogTimer >= WATCHDOG_TIMEOUT) {
+    if(!isEStopped) {
+      nh->logerror("Lost connectivity . . .");
+      eStopTractor();
+    }
+  }
+}
+
+
+char checkUserInput() {
+  // Get most recent user-sent input (if any), reset global var if necessary.
+  auto outMsg = usrMsg;
```

Comment:
again, should usr be initialized or is it ok?

Reply:
Probably a race condition, fixed.